src/program.ts

```diff
@@ -490,40 +490,33 @@ export class Program extends DiagnosticEmitter {
     }
 
     // queued imports should be resolvable now through traversing exports and queued exports
-    for (let i = 0; i < queuedImports.length;) {
-      let queuedImport = queuedImports[i];
+    for (const queuedImport of queuedImports) {
```

Comment:
Looks good, but can we keep the plain for-loop here? Just to be sure, because for-of isn't yet supported when compiling to WASM (somewhat due to the lack of iterators), even though we don't compile the compiler itself to WASM yet.